extension/goext/error_test.go

```diff
@@ -71,21 +73,85 @@ var _ = Describe("Error", func() {
 		})
 
 		It("Should return full error stack on request", func() {
-			Expect(fmt.Sprintf("%s", errSingle.ErrorStack())).To(Equal("HTTP 500 (Internal Server Error) at first.go:100: it does not work"))
-			Expect(fmt.Sprintf("%s", errDouble.ErrorStack())).To(Equal(
+			Expect(fmt.Sprintf("%s", errSingle.ErrorStack())).To(ContainSubstring("HTTP 500 (Internal Server Error) at first.go:100: it does not work"))
+			Expect(fmt.Sprintf("%s", errDouble.ErrorStack())).To(ContainSubstring(
 				`HTTP 503 (Service Unavailable) at one.go:1000 from
   <- HTTP 305 (Use Proxy) at two.go:2000: nothing here`))
 		})
 
 		It("Should capture correct stack", func() {
-			error := goext.NewErrorInternalServerError(fmt.Errorf("test error"))
+			error := goext.NewErrorInternalServerError(errors.New("test error"))
 			_, _, line, _ := runtime.Caller(0)
 			Expect(error.Origin).To(HaveSuffix(fmt.Sprintf("github.com/cloudwan/gohan/extension/goext/error_test.go:%d", line-1)))
 		})
 
+		It("Should capture full stack for wrapped errors", func() {
+			var innerLine int
+			innerFunc := func() *goext.Error {
+				_, _, line, _ := runtime.Caller(0)
+				innerLine = line + 2
+				return goext.NewErrorBadGateway(fmt.Errorf("test error"))
+			}
+
+			var middleLine int
+			middleFunc := func() error {
+				_, _, line, _ := runtime.Caller(0)
+				middleLine = line + 2
+				return innerFunc()
+			}
+
+			var outerLine int
+			outerFunc := func() *goext.Error {
+				_, _, line, _ := runtime.Caller(0)
+				outerLine = line + 2
+				return goext.NewErrorBadRequest(middleFunc())
+			}
+
+			_, _, line, _ := runtime.Caller(0)
+			callLine := line + 2
+			err := outerFunc()
+			Expect(err.ErrorStack()).To(ContainSubstring(fmt.Sprintf("error_test.go:%d", innerLine)))
+			Expect(err.ErrorStack()).To(ContainSubstring(fmt.Sprintf("error_test.go:%d", middleLine)))
+			Expect(err.ErrorStack()).To(ContainSubstring(fmt.Sprintf("error_test.go:%d", outerLine)))
+			Expect(err.ErrorStack()).To(ContainSubstring(fmt.Sprintf("error_test.go:%d", callLine)))
```

Comment:
Also should we check BadGateway here?

Reply:
good idea. also a similar assertion added to `Should capture full stack for built-in errors`